controllers/configentry_controller.go

```diff
@@ -0,0 +1,299 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	consulv1alpha1 "github.com/hashicorp/consul-k8s/api/v1alpha1"
+	"github.com/hashicorp/consul-k8s/namespaces"
+	capi "github.com/hashicorp/consul/api"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	FinalizerName    = "finalizers.consul.hashicorp.com"
+	ConsulAgentError = "ConsulAgentError"
+)
+
+// Reconciler is implemented by CRD-specific reconcilers. It is used by
+// ConfigEntryReconciler to abstract CRD-specific reconcilers.
+type Reconciler interface {
```

Comment:
Can we rename this? `controller-runtime` has an interface with the same name which is what the Reconcilers actually implement and this  could lead to confusion.

Reply:
What do you think about calling them `Controller`s? We already call the file that.

Reply:
SGTM!!

Reply:
Done.